src/Functions/FunctionSpaceFillingCurve.h

```diff
@@ -47,6 +47,10 @@ class FunctionSpaceFillingCurveEncode: public IFunction
             vector_start_index = 1;
             const auto * type_tuple = typeid_cast<const DataTypeTuple *>(arguments[0].get());
             auto tuple_size = type_tuple->getElements().size();
+            if (tuple_size == 0)
+                throw Exception(ErrorCodes::TOO_FEW_ARGUMENTS_FOR_FUNCTION,
```

Comment:
```suggestion
                throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
```

Reply:
Hm, but why it goes out of tuple bounds, with empty arguments, I think we should fix that place

Reply:
Also does `mortonDecode` has the same bug?